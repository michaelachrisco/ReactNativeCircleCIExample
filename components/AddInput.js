//AddInput.js

import React, { useState } from "react";
import {View, TextInput, Text, TouchableOpacity} from 'react-native'
import {Alert} from 'react-native';
import styled from "styled-components";

export default function AddInput() {
  const [value, setValue] = useState("");

  const onChangeText = (text) => {
    setValue(text);
  };

  

  return (
    <View>
      <View>
        <TextInput placeholder="Add Task..." onChangeText= 
         {onChangeText} />
      </View>
      <TouchableOpacity
      title='Add task...'
        onPress={() => {
            Alert.alert('button clicked')
        }}
      >
        <Text>Submit</Text>
      </TouchableOpacity>
    </View>
  );

  //styles
const ComponentContainer = styled.View`
flex-direction: row;
`;

const InputContainer = styled.View`
flex-direction: row;
border-radius: 10px;
`;

const Input = styled.TextInput`
font-size: 20px;
background-color: white;
width: 300px;
margin-right: 20px;
padding: 10px;
margin-bottom: 20px;
border-radius: 10px;
`;

const SubmitButton = styled.TouchableOpacity`
width: 50px;
justify-content: center;
align-items: center;
background-color: whitesmoke;
margin-bottom: 20px;
border-radius: 50px;
`;
}
