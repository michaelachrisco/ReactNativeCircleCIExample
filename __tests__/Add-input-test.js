import "react-native";
import React from "react";
import AddInput from "../components/AddInput";

import { fireEvent, render, waitFor } from "@testing-library/react-native";
import { Alert } from 'react-native';
jest.spyOn(Alert, 'alert');

it("Renders Message", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<AddInput />);

  const button = getByText("Submit");
  fireEvent.press(button);

  expect(Alert.alert).toHaveBeenCalledWith("button clicked");
  expect(toJSON()).toMatchSnapshot();
});
