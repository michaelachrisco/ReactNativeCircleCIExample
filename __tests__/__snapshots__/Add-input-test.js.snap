// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Message 1`] = `
<View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Add Task..."
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Submit
    </Text>
  </View>
</View>
`;
